BUILDING.MD

```diff
@@ -0,0 +1,226 @@
+# Building TP Randomizer
+
+For more information about developing for the Randomizer, please check the [wiki](https://wiki.zeldatp.net/Dev).
+
+## Table of contents
+
+- [Prerequisites](#prerequisites)
+  - [Windows](#windows)
+  - [Non-Windows](#non-windows)
+- [Building](#building)
+  - [Building the Randomizer](#building-the-randomizer)
+- [Testing in Dolphin](#testing-in-dolphin)
+  - [Loading the Randomizer](#loading-the-randomizer)
+
+## Prerequisites
+
+### Windows
+
+Install [devkitPro](https://devkitpro.org/wiki/Getting_Started) which is used to compile the Randomizer code to PowerPC instructions which the GameCube and Wii understand.
+
+1. Download the graphical installer by following the above link.
+
+1. Within the installer, the following selections are recommended:
+
+   - "Download and install/install from downloaded files"
+   - "Remove downloaded files"
+   - For components to install, you must leave "Gamecube Development" checked.
+     - It is recommended that you also leave "Wii Development" checked since there are plans for the Randomizer to support Wii versions in the future.
+       You can uncheck the others options.
+   - You can leave the Destination Folder as the default (likely `C:\devkitPro`).
+   - You can leave the Start Menu folder as the default.
+   - Click "Install".
+     Installation may take a few minutes.
+
+1. Once installation finishes, navigate to the devkitPro folder (likely `C:\devkitPro`).
+
+1. Change to the `msys2` folder, then run `msys2.exe` by double-clicking.<br>
+   _([MSYS2](#https://www.msys2.org) is a collection of tools and libraries providing you with an easy-to-use environment for building, installing and running native Windows software.)_
+
+1. Within the `msys2` shell, run the following to install `python3` and `git` for the msys2 environment:
+
+   ```
+   pacman -S python3 git
+   ```
+
+   - You can ignore any warnings about conflicting packages and proceed with installation.
+
+**NOTE: it is recommended that you use `msys2.exe` as your shell when building the Randomizer.**
+
+### Non-Windows
+
+If you would like to provide instructions for non-Windows environments, we would be happy to review your pull request. Thanks!
+
+## Building
+
+The commands you should run to clone the Randomizer repository will vary depending on whether or not you plan to create any pull requests:
```

Comment:
I can add instructions for Arch (and alike; should be similar on most Linux distributions)